app/controllers/submit.js

```diff
@@ -899,15 +1019,70 @@ export default Ember.Controller.extend(Analytics, BasicsValidations, NodeActions
             const node = this.get('node');
             this.set('savingPreprint', true);
             this.toggleProperty('shareButtonDisabled');
-            model.set('isPublished', true);
+
+            let log = null;
+            if (this.get('moderationType')) {
+                log = this.get('store').createRecord('reviewLog', {
+                   action: 'submit',
+                   reviewable: this.get('model')
+                });
+            } else {
+                model.set('isPublished', true);
+            }
             node.set('public', true);
 
+            if (log) {
+                return log.save()
```

Comment:
I think it would make more sense to save the log last, after all the updates to the preprint and node (especially setting the node public) go through successfully.